protocols/discovery/src/substream.rs

```diff
@@ -129,16 +129,33 @@ impl SubstreamValue {
         substream: Substream,
         max_known: usize,
         query_cycle: Option<Duration>,
+        global_ip_only: bool,
     ) -> SubstreamValue {
         let session_id = substream.stream.session_id;
         let mut pending_messages = VecDeque::default();
         debug!("direction: {:?}", direction);
         let mut addr_known = AddrKnown::new(max_known);
-        let remote_addr = if direction.is_outbound() {
+        let is_reachable = !global_ip_only
```

Comment:
Can we keep the same code in this block and move the `global_ip_only` checking to line 242 ?